digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/TrafficSignService.scala

```diff
@@ -155,4 +170,67 @@ class TrafficSignService(val roadLinkService: RoadLinkService) extends PointAsse
       persistedStop.linkSource)
   }
 
+  def getAssetValidityDirection(bearing: Int): Option[Int] = {
+    if(bearing >= 270 || bearing < 90){
+      Some(AgainstDigitizing.value)
+    }else{
+      Some(TowardsDigitizing.value)
+    }
+  }
+
+  def getTrafficSignValidityDirection(assetLocation: Point, geometry: Seq[Point]): Int = {
+
+    val mValue = GeometryUtils.calculateLinearReferenceFromPoint(Point(assetLocation.x, assetLocation.y, 0), geometry)
+    val roadLinkPoint = GeometryUtils.calculatePointFromLinearReference(geometry, mValue)
+    val linkBearing = GeometryUtils.calculateBearing(geometry)
+
+    val lonDifference = assetLocation.x - roadLinkPoint.get.x
+    val latDifference = assetLocation.y - roadLinkPoint.get.y
+
+
+    if((latDifference <= 0 && linkBearing <= 90) || (latDifference >= 0 && linkBearing > 270)) {
+      TowardsDigitizing.value
+    }else{
+      AgainstDigitizing.value
+    }
+  }
+
+  def getAssetBearing(validityDirection: Int, geometry: Seq[Point] ): Int = {
+    val linkBearing = GeometryUtils.calculateBearing(geometry)
+    GeometryUtils.calculateActualBearing(validityDirection, Some(linkBearing)).get
+  }
+
+  private def generateProperties(tRTrafficSignType: TRTrafficSignType, value: Any) = {
+    val trafficType = tRTrafficSignType.trafficSignType
```

Comment:
Maybe instead of value: Any this method could receive value: Option[Int] and convert only one time to string

Reply:
Changed to one time toString, but the Option[Int] alternative will return either Some(Int) or None, which looks a bit clumsy as a property